src/message.ts

```diff
@@ -1,13 +1,20 @@
 import {CommitHash, nullCommitHash} from './commit-hash'
 import {MessageKey} from './message-key'
+import {NO_JOB_ID} from './job'
 
 export abstract class Message {
   payload: string
   jobRef: CommitHash
+  id: number
 
-  constructor(payload: string, jobRef: CommitHash = nullCommitHash()) {
+  constructor(
+    payload: string,
+    jobRef: CommitHash = nullCommitHash(),
+    id: number
```

Comment:
@yeraydavidrodriguez I'm not sure how this work when you want to use the default CommitHash, I think you have to call it like this:

```
new Message('payload', undefined, 42)
```

If the types change It could be even more confusing. I would reorder them to:

```
constructor(
  id: jobId,
  payload: string,
  jobRef: CommitHash = nullCommitHash(),
```

Reply:
Good point. Applied.